### PR TITLE
Gravatar - switch to Icon template group

### PR DESCRIPTION
**Pull request by altern8tif**
- amend normalize and templates in gravatar.js
- remove unnecessary css
- add test cases
- remove content.handlebars

- address #1864

### CODE REVIEW

**Comment by altern8tif** [comment]
screen shot 2015-06-14 at 12 41 38 pm

**Comment by moollaza** [comment]
@altern8tif this is looking great.

@abeyang this is now using the icon template. I suspect you and Chris will want the links to be grey? If so, we should change the template CSS ;)

**Comment by altern8tif** [comment]
Added `variants` attributes so that custom CSS can be removed.

File: share/spice/gravatar/gravatar.js [hunk 1]
```diff
@@ -20,77 +16,64 @@
...
                         return entry.name.givenName + " " + entry.name.familyName;
                     }
                     return entry.displayName;
                 }
+                // Get current location
+                function getCurrentLocation(entry) {
+                    if (entry.currentLocation) {
+                        return entry.currentLocation;
+                    } else {
+                        return "No Location Specified";
```
Comment (moollaza) [inline share/spice/gravatar/gravatar.js]: Let's rather not display anything if the info isn't there. Better to `return null`

File: share/spice/gravatar/gravatar.js [hunk 2]
```diff
@@ -20,77 +16,64 @@
...
+                    var accounts = [];
+                    if (entry.accounts) {
+                        $.each(entry.accounts, function() {
+                            accounts.push({
+                                text: DDG.capitalize(this.shortname),
+                                href: this.url
+                            });
+                        });
+                    } else {
+                        accounts = ["Additional Information Unavailable"];
```
Comment (moollaza) [inline share/spice/gravatar/gravatar.js]: Same here, let's not show anything if it doesn't exist.

**Comment by moollaza** [comment]
@abeyang this one is ready for a design pass, it's using the Icon template.

File: share/spice/gravatar/gravatar.js (hunk 1 above)
Comment (altern8tif) [inline share/spice/gravatar/gravatar.js]: OK done!

File: share/spice/gravatar/gravatar.js (hunk 2 above)
Comment (altern8tif) [inline share/spice/gravatar/gravatar.js]: OK done!

**Comment by abeyang** [comment]
@altern8tif , it's looking pretty good! However, can we make a few more tweaks, such that:

- We should show the gravatar handle, not just the social networks the user is on
- Instead of showing *all* the social networks, can we just show the 3 main ones, and have a "X More" link at the end (which points to their main gravatar site)?
- Also, can we utilize a smaller avatar size -- let's use the **medium icon** and **medium-sized title**, found on our [styleguide](https://duckduckgo.com/styleguide)

In the end, it should look like this (don't worry about the link colors -- we'll fix that):

gravatar

Thanks!

**Comment by altern8tif** [comment]
@abeyang how about showing the social network icon/favicon next to the handle?  This helps to identify which social network the handle belongs to... As some pple might have the same handles for different social networks.

**Comment by abeyang** [comment]
@altern8tif I don't think that's necessary, because the query itself is for gravatar. It's similar when you do a [twitter](https://duckduckgo.com/?q=twitter+matt&ia=social) search. Besides, the favicon is already showing at the bottom, so I think we're good.

**Comment by altern8tif** [comment]
@abeyang Ah I think I misunderstood you the first time.  Sure, will make the changes as per your attached screenshot :smiley:

**Comment by altern8tif** [comment]
@abeyang Let me know if this works...

**Comment by moollaza** [comment]
@altern8tif LGTM :+1:

**Comment by moollaza** [comment]
@altern8tif crap, just found a bug. Try "gravatar bob" we're trying to pass an object as as string:
gravatar_bob_at_duckduckgo

**Comment by moollaza** [comment]
This is one is even worse:
gravatar_tom_at_duckduckgo

Let's use a relevancy block to require that the user at least has a valid username so we have something to show!

File: share/spice/gravatar/gravatar.js [hunk 3]
```diff
@@ -20,77 +16,71 @@
...
+                    }
+                }
+                // Get array of social media accounts
+                function getAccounts(entry) {
+                    var accounts = [];
+                    var maxAccounts = 4;
+                    if (entry.accounts) {
+                        $.each(entry.accounts, function() {
+                            accounts.push({
+                                text: DDG.capitalize(this.shortname),
```
Comment (moollaza) [inline share/spice/gravatar/gravatar.js]: Seems like we need to make sure these are defined

**Comment by altern8tif** [comment]
@moollaza Sorry my bad... I should have done further testing on other usernames.

The thing about Gravatar is that there are numerous accounts (like [tom](https://en.gravatar.com/tom.json)) where the account is totally empty... It'll be funny if everything were empty.

Should I default to showing the same info as the existing live Gravatar IA (see screenshot)?

screen shot 2015-07-04 at 2 29 36 am

**Pull request merged**